### PR TITLE
Merge master H(bb) ntuples

### PR DESCRIPTION
**Pull request by DylanHsu**

### CODE REVIEW

**Comment by DylanHsu** [comment]
with the ad-hoc nominal jet pt fix:
```root [1] events->Scan("jotPt:nJot:nJotMax")
***********************************************************
*    Row   * Instance *     jotPt *      nJot *   nJotMax *
***********************************************************
*        0 *        0 * 45.064304 *         3 *         4 *
*        0 *        1 * 22.115665 *         3 *         4 *
*        0 *        2 * 21.369323 *         3 *         4 *
*        0 *        3 * 17.484472 *         3 *         4 *
*        1 *        0 * 96.864944 *         5 *         8 *
*        1 *        1 * 55.299877 *         5 *         8 *
*        1 *        2 * 50.516391 *         5 *         8 *
*        1 *        3 * 45.942451 *         5 *         8 *
*        1 *        4 * 29.488508 *         5 *         8 *
*        1 *        5 * 29.488508 *         5 *         8 *
*        1 *        6 * 19.684864 *         5 *         8 *
*        1 *        7 * 12.955325 *         5 *         8 *
*        2 *        0 * 236.08477 *         8 *         9 *
*        2 *        1 * 171.67314 *         8 *         9 *
*        2 *        2 * 128.02024 *         8 *         9 *
*        2 *        3 * 80.886383 *         8 *         9 *
*        2 *        4 * 57.583549 *         8 *         9 *
*        2 *        5 * 34.143867 *         8 *         9 *
*        2 *        6 * 33.300884 *         8 *         9 *
*        2 *        7 * 33.921936 *         8 *         9 *
*        2 *        8 * 16.652719 *         8 *         9 *
*        3 *        0 * 97.253318 *         2 *         2 *
*        3 *        1 * 53.848445 *         2 *         2 *
*        4 *        0 * 89.832672 *         4 *         5 *
*        4 *        1 * 63.308261 *         4 *         5 *
```

without the ad-hoc nominal jet pt fix:

```***********************************************************
*    Row   * Instance *     jotPt *      nJot *   nJotMax *
***********************************************************
*        0 *        0 * 45.064304 *         3 *         4 *
*        0 *        1 * 22.115665 *         3 *         4 *
*        0 *        2 * 21.369323 *         3 *         4 *
*        0 *        3 *       -99 *         3 *         4 *
*        1 *        0 * 96.864944 *         5 *         8 *
*        1 *        1 * 55.299877 *         5 *         8 *
*        1 *        2 * 50.516391 *         5 *         8 *
*        1 *        3 * 45.942451 *         5 *         8 *
*        1 *        4 * 29.488508 *         5 *         8 *
*        1 *        5 *       -99 *         5 *         8 *
*        1 *        6 *       -99 *         5 *         8 *
*        1 *        7 *       -99 *         5 *         8 *
*        2 *        0 * 236.08477 *         8 *         9 *
*        2 *        1 * 171.67314 *         8 *         9 *
*        2 *        2 * 128.02024 *         8 *         9 *
*        2 *        3 * 80.886383 *         8 *         9 *
*        2 *        4 * 57.583549 *         8 *         9 *
*        2 *        5 * 34.143867 *         8 *         9 *
*        2 *        6 * 33.300884 *         8 *         9 *
*        2 *        7 * 33.921936 *         8 *         9 *
*        2 *        8 *       -99 *         8 *         9 *
*        3 *        0 * 97.253318 *         2 *         2 *
*        3 *        1 * 53.848445 *         2 *         2 *
*        4 *        0 * 89.832672 *         4 *         5 *
*        4 *        1 * 63.308261 *         4 *         5 *
```

**Comment by DylanHsu** [comment]
The ad hoc fix is also not correct for jets beyond `nJot` due to the potential change in ordering in different JES scenarios. See example in row 1 of last message. Finding a real fix now.

File: Flat/interface/JetsMods.h [hunk 1]
```diff
@@ -208,14 +208,14 @@ namespace pa {
           jerV = "Spring16_25nsV10";
           eraGroups = {"BCD","EF","G","H"};
           spacer = "";
-          csvL = 0.5426; csvM = 0.8484;
```
Comment (sidnarayanan) [inline Flat/interface/JetsMods.h:211]: Can't change these WPs.

File: Flat/src/BTagTools.cc [hunk 2]
```diff
@@ -52,14 +52,19 @@ BTagCorrs::BTagCorrs(TString dirPath, const Analysis& analysis, GeneralTree& gt_
 
       if (analysis.year==2016) {
         if (analysis.useCMVA) 
-          reshaper_calib.reset(new BTagCalibration("csvv2", 
-                                                   (dirPath+"moriond17/CSVv2_Moriond17_B_H.csv").Data()));
-        else
```
Comment (sidnarayanan) [inline Flat/src/BTagTools.cc]: Stop breaking my analyses.

File: Flat/src/CommonMods.cc [hunk 3]
```diff
@@ -284,8 +284,8 @@ void TriggerMod::checkEle32()
 void TriggerMod::do_execute()
 {
   for (unsigned iT = 0; iT != kNTrig; ++iT) {
-    if (analysis.hbb && analysis.year==2017 && iT==kSingleEleTrig) 
-      checkEle32();
+    //if (analysis.hbb && analysis.year==2017 && iT==kSingleEleTrig) 
+    //  checkEle32();
```
Comment (sidnarayanan) [inline Flat/src/CommonMods.cc]: If unused, delete instead of comment.

File: Flat/src/JetsMods.cc [hunk 4]
```diff
@@ -164,16 +164,23 @@ void JetMod::do_execute()
...
+        if (analysis.hbb && jet.puid < utils.getCorr(cJetLoosePUID, aeta,min(39.99f,pt)))
+          jw.isPileupJet = true;
+      }
+      if (jw.nominal->isLep || jw.nominal->isPho || jw.nominal->isPileupJet)
         continue;
       if ((analysis.vbf || analysis.hbb) && !jet.loose)
         continue;
 
-      float csv = centralOnly(analysis.year==2016 ? jet.csv : jet.deepCSVb, aeta);
+      float csv = centralOnly(jet.deepCSVb + jet.deepCSVbb, aeta);
```
Comment (sidnarayanan) [inline Flat/src/JetsMods.cc]: Again.

File: Flat/src/Module.cc [hunk 5]
```diff
@@ -113,8 +113,8 @@ void ConfigMod::set_inputBranches()
 
     if (cfg.isData || analysis.mcTriggers) {
       bl.push_back("triggers");
-      if (analysis.hbb && analysis.year==2017)
-        bl.push_back("triggerObjects");
+      //if (analysis.hbb)
+      //  bl.push_back("triggerObjects");
```
Comment (sidnarayanan) [inline Flat/src/Module.cc]: Delete.

File: Flat/test/test.py [hunk 6]
```diff
@@ -25,11 +25,22 @@
...
 a.recalcECF = True
 a.varyJESTotal = True
+
+#a = wlnhbb2017()
+#a = wlnhbb()
+#a = breg()
+# a.bjetDeepReg = True
+# a.bjetRegTrain = True
+#a.processType = root.pa.kH
+#a.processType = root.pa.kW
```
Comment (sidnarayanan) [inline Flat/test/test.py]: Delete.

File: T3/bin/buildMergedInputs.sh [hunk 7]
```diff
@@ -37,7 +37,7 @@ cp -v ${WD}/local.cfg ${WD}/local_all.cfg
 cd $CMSSW_BASE/
 if [[ $doTar == 1 ]]; then
   PInfo -n "$myName" "Tarring up CMSSW..."
-  tar --exclude-vcs -chzf cmssw.tgz src python biglib bin lib objs test external # h = --dereference symlinks
+  tar --exclude-vcs -chzf cmssw.tgz src/Panda* python biglib bin lib objs test external # h = --dereference symlinks
```
Comment (sidnarayanan) [inline T3/bin/buildMergedInputs.sh]: C'mon dude.

File: T3/merging/bigsamples.sh [hunk 8]
```diff
@@ -0,0 +1,23 @@
+#threshold=4718592 # 4.5 GB
```
Comment (sidnarayanan) [inline T3/merging/bigsamples.sh:1]: What's the purpose of checking this in if we have another tool to the same thing?

File: T3/merging/merge.py [hunk 9]
```diff
@@ -8,17 +8,21 @@
...
 parser.add_argument('--cfg', type=str, default='common')
 parser.add_argument('--skip_missing', action='store_true')
+parser.add_argument('--make_chunks', action='store_true')
 parser.add_argument('arguments', type=str, nargs='+')
 args = parser.parse_args()
 arguments = args.arguments
 VERBOSE = not args.silent
 skip_missing = args.skip_missing
+make_chunks = args.make_chunks
+#make_chunks = True
```
Comment (sidnarayanan) [inline T3/merging/merge.py]: Delete.

File: T3/merging/merge.py [hunk 10]
```diff
@@ -186,8 +192,22 @@ def merge(shortnames,mergedname):
         args[pd] = [pd]
 
 for pd in args:
+    disk="tmp"
+    checkSizeCmd = "du -c {0}/{1}_*.root|grep total|sed 's/\t\+total//g'".format(environ['SUBMIT_OUTDIR'],pd)
```
Comment (sidnarayanan) [inline T3/merging/merge.py]: Please rewrite.

File: T3/merging/merge.py [hunk 11]
```diff
@@ -186,8 +192,22 @@ def merge(shortnames,mergedname):
...
+    unmergedSize = int(subprocess.check_output(['bash','-c', checkSizeCmd]))
+    if unmergedSize > 15*1024*1024: # 15 GB
+        disk="scratch5"
+    split_dir = '/%s/%s/split/%s/'%(disk, user, submit_name)
+    merged_dir = '/%s/%s/merged/%s/'%(disk, user, submit_name)
```
Comment (sidnarayanan) [inline T3/merging/merge.py]: What happens if these don't exist?

File: T3/merging/merge.py [hunk 12]
```diff
@@ -36,6 +40,8 @@
 cfg = __import__(args.cfg)
 
 Load('Normalizer')
+if make_chunks:
+  Load('splitPandaExpress')
```
Comment (sidnarayanan) [inline T3/merging/merge.py]: Where is this being loaded from?

**Comment by sidnarayanan** [comment]
This PR needs work.

**Comment by sidnarayanan** [comment]
Also I think you did `git add *` at some point when merging changes and checked in a bunch of deleted code. Undo that as well.

**Review by sidnarayanan — CHANGES_REQUESTED** [review]

File: Flat/interface/JetsMods.h (hunk 1 above)
Comment (DylanHsu) [reply Flat/interface/JetsMods.h:211]: already fixed

**Comment by DylanHsu** [comment]
I removed the chunking code from the merger, and made the disk decision more python esque

**Review by sidnarayanan — APPROVED** [review]

**Pull request merged**